### ISSUE TITLE
PasswordField should not contain real password

### ISSUE THREAD

**Issue by fredl11** [vaadin/framework#10544]
This behavior is working as described [in the documentation](https://vaadin.com/docs/v8/framework/components/components-passwordfield.html):

> Unless the server connection is encrypted with a secure connection, such as HTTPS, the input is transmitted in clear text and may be intercepted by anyone with low-level access to the network.

I think it would be good practice to not reveal the actual password to the browser. The PasswordField should handle this transparently.

This was observed with Vaadin 8.2.1, but since it is working as documented, every version will be affected.

**Comment by Legioth** [comment]
Changing `PasswordField` to not send the string passed to `setValue` to the browser is an interesting idea, though I suspect the actual benefit is quite small. The reason for this is that in typical cases, `PasswordField` is used for transmitting password from the user and to the server (e.g. login or changing their password) and not in the opposite direction.

I would even claim that an application that has an actual value to pass to `setValue()` is typically doing something wrong since that means that they're actually storing the user's password (or a decryptable representation) instead of only storying a properly salted hash of the original password.

**Comment by fredl11** [comment]
I think your observation is correct. The behavior I propose could also be achieved manually today by implementing a getter/setter combination which does not reveal the password but uses a place holder value instead.

However, your second claim applies in my opinion only to applications that do not need the real password. Our application requires the actual password for authentication against SMTP servers or database servers. The `PasswordField` is used in the administrative interface for editing these connections.

**Comment by focbenz** [comment]
We are also affected by this issue and as a result have removed all PasswordFields from the application.
Every user can intercept the password with the browser debugger (e.g. pressing "F12") as long as the PasswordField is displayed on the screen.

**Comment by Legioth** [comment]
Good point about storing passwords for other purposes than end-users' login details. This is indeed something that could be considered, although I'm afraid it might not be on the top of our own priority lists.

**Comment by Legioth** [comment]
Should be quite easily implemented with a separate server-side field for the real value and then storing a random bogus value in the shared state, although you might then also have to override some logic inherited from `AbstractTextField`. In addition to this, it would also good to have a two-state configuration mode to let users opt in or out from this behavior since it might affect integration with e.g. TestBench or some extensions.

**Comment by focbenz** [comment]
Please don't stale ... I think other people might still be affected using the PasswordField for display purposes.

**Comment by Legioth** [comment]
No changes on this front in newer versions. Seems like the warning in the documentation just wasn't picked up when documentation for newer versions was rewritten from scratch.

I created https://github.com/vaadin/docs/issues/3588 to get this addressed in the current documentation.

**Comment by fabst2w** [comment]
For me a combination of custom getter and setter works fine with Vaadin 24, as fredl11 mentioned:
The system generates a dummy password that is ok to all password validators.
The custom getter usually returns this dummy password so that the browser usually only gets this in plain text.
When a user clicks into the password field, the field cleans itself to enforce entering a new one.
The custom setter checks whether the newly entered password is different from the dummy password and only then sets this new password value at the domain object.
